crates/polars-lazy/src/physical_plan/planner/lp.rs

```diff
@@ -165,10 +165,11 @@ pub fn create_physical_plan(
             Ok(Box::new(executors::SliceExec { input, offset, len }))
         },
         Selection { input, predicate } => {
+            let input_schema = lp_arena.get(input).schema(lp_arena).into_owned();
             let input = create_physical_plan(input, lp_arena, expr_arena)?;
             let mut state = ExpressionConversionState::default();
             let predicate =
-                create_physical_expr(predicate, Context::Default, expr_arena, None, &mut state)?;
+                create_physical_expr(predicate, Context::Default, expr_arena, Some(&input_schema), &mut state)?;
```

Comment:
I'm not sure if it's due to some reason I don't know that we didn't pass `input_schema` here. 🤔